### PR TITLE
added the data transformation process in the training pipeline

### PR DESCRIPTION
**Pull request by Rigved-Harmalkar**
Added transformation, evaluation and deployment using Fast API in the application.

### CODE REVIEW

File: Hate/components/data_transformation.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import nltk
+from nltk.corpus import stopwords
+nltk.download('stopwords')
+from sklearn.model_selection import train_test_split
+from Hate.logger import logging 
+from Hate.exception import CustomException
+from Hate.entity.config_entity import DataTransformationConfig
+from Hate.entity.artifact_entity import DataIngestionArtifacts, DataTransformationArtifacts
+
+
```
Comment (ashish-k-r) [inline Hate/components/data_transformation.py]: Please write a description of what this class aims to do.

File: Hate/components/data_transformation.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+class DataTransformation:
+    def __init__(self,data_transformation_config: DataTransformationConfig,data_ingestion_artifacts:DataIngestionArtifacts):
+        self.data_transformation_config = data_transformation_config
+        self.data_ingestion_artifacts = data_ingestion_artifacts
+        
+        
+
```
Comment (ashish-k-r) [inline Hate/components/data_transformation.py]: Add descriptions for the methods calling out their functionalities.

File: Hate/components/model_trainer.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+from Hate.logger import logging
+from Hate.constants import *
+from Hate.exception import CustomException
+from sklearn.model_selection import train_test_split
+from keras.preprocessing.text import Tokenizer
+from keras.utils import pad_sequences
+from Hate.entity.config_entity import ModelTrainerConfig
+from Hate.entity.artifact_entity import ModelTrainerArtifacts,DataTransformationArtifacts
+from Hate.ml.model import ModelArchitecture
+
```
Comment (ashish-k-r) [inline Hate/components/model_trainer.py]: Same as previous comments.

File: Hate/entity/artifact_entity.py [hunk 4]
```diff
@@ -4,4 +4,15 @@
 @dataclass
 class DataIngestionArtifacts:
     imbalance_data_file_path: str
-    raw_data_file_path: str
\ No newline at end of file
+    raw_data_file_path: str
+    
+
```
Comment (ashish-k-r) [inline Hate/entity/artifact_entity.py]: Same as previous comments.

File: Hate/entity/config_entity.py [hunk 5]
```diff
@@ -11,4 +11,39 @@ def __init__(self):
         self.DATA_ARTIFACTS_DIR: str = os.path.join(self.DATA_INGESTION_ARTIFACTS_DIR,DATA_INGESTION_IMBALANCE_DATA_DIR)
         self.NEW_DATA_ARTIFACTS_DIR: str = os.path.join(self.DATA_INGESTION_ARTIFACTS_DIR,DATA_INGESTION_RAW_DATA_DIR)
         self.ZIP_FILE_DIR = os.path.join(self.DATA_INGESTION_ARTIFACTS_DIR)
-        self.ZIP_FILE_PATH = os.path.join(self.DATA_INGESTION_ARTIFACTS_DIR,self.ZIP_FILE_NAME)
\ No newline at end of file
+        self.ZIP_FILE_PATH = os.path.join(self.DATA_INGESTION_ARTIFACTS_DIR,self.ZIP_FILE_NAME)
+        
+
```
Comment (ashish-k-r) [inline Hate/entity/config_entity.py]: Same as previous comments.

File: Hate/ml/model.py [hunk 6]
```diff
@@ -0,0 +1,24 @@
+# Creating model architecture.
+from Hate.entity.config_entity import ModelTrainerConfig
+from keras.models import Sequential
+from keras.optimizers import RMSprop
+from keras.callbacks import EarlyStopping, ModelCheckpoint
+from keras.layers import LSTM,Activation,Dense,Dropout,Input,Embedding,SpatialDropout1D
+from Hate.constants import *
+
```
Comment (ashish-k-r) [inline Hate/ml/model.py]: Same as previous comments.

File: Hate/pipeline/train_pipeline.py [hunk 7]
```diff
@@ -2,8 +2,18 @@
 from Hate.logger import logging
 from Hate.exception import CustomException
 from Hate.components.data_ingestion import DataIngestion
-from Hate.entity.config_entity import (DataIngestionConfig)
-from Hate.entity.artifact_entity import (DataIngestionArtifacts)
```
Comment (ashish-k-r) [inline Hate/pipeline/train_pipeline.py:6]: Same as previous comments.

**Review by ashish-k-r — CHANGES_REQUESTED** [review]
Change the name of the project folder, and add descriptions to any class or method that you write.

**Review by ashish-k-r — APPROVED** [review]

File: Hate/components/data_transformation.py (hunk 2 above)
Comment (Rigved-Harmalkar) [reply Hate/components/data_transformation.py]: Appropriate changes have been made.

File: Hate/components/model_trainer.py (hunk 3 above)
Comment (Rigved-Harmalkar) [reply Hate/components/model_trainer.py]: Appropriate changes have been made.

File: Hate/entity/config_entity.py (hunk 5 above)
Comment (Rigved-Harmalkar) [reply Hate/entity/config_entity.py]: Appropriate changes have been made.

File: Hate/entity/artifact_entity.py (hunk 4 above)
Comment (Rigved-Harmalkar) [reply Hate/entity/artifact_entity.py]: Appropriate changes have been made.

File: Hate/ml/model.py (hunk 6 above)
Comment (Rigved-Harmalkar) [reply Hate/ml/model.py]: Appropriate changes have been made.

File: Hate/pipeline/train_pipeline.py (hunk 7 above)
Comment (Rigved-Harmalkar) [reply Hate/pipeline/train_pipeline.py:6]: Appropriate changes have been made.

**Pull request merged**